app/workers/commit_monitor_handlers/batch/github_pr_commenter/diff_content_checker.rb

```diff
@@ -0,0 +1,48 @@
+module CommitMonitorHandlers::Batch
+  class GithubPrCommenter::DiffContentChecker
+    include Sidekiq::Worker
+    sidekiq_options :queue => :miq_bot_glacial
+
+    include BatchEntryWorkerMixin
+    include BranchWorkerMixin
+
+    def perform(batch_entry_id, branch_id, _new_commits)
+      return unless find_batch_entry(batch_entry_id)
+      return unless find_branch(branch_id, :pr)
+      complete_batch_entry({:result => process_lines})
+    end
+
+    private
+
+    def process_lines
+      @offenses = []
+
+      branch.git_service.diff.with_each_line do |line, _parent_hunk, parent_patch|
+        next unless line.addition?
+        check_line(line, parent_patch)
+      end
+
+      @offenses
+    end
+
+    def check_line(line, patch)
+      Settings.diff_content_checker.each do |offender, options|
+        next if options.except.try(:any?) { |except| patch.delta.new_file[:path].start_with?(except) }
+
+        if options.type == :regexp
+          add_offense(offender, options, patch, line) if Regexp.new(offender.to_s).match(line.content)
+        else
+          add_offense(offender, options, patch, line) if line.content.downcase.include?(offender.to_s)
```

Comment:
This can be simplified, but it's ok for now...

```ruby
regexp = options.type == :regexp ? Regexp.new(offender.to_s) : /#{Regexp.escape(offender)}/i
add_offense(offender, options, patch, line) if regexp.match(line.content)
```